internal/command/test.go

```diff
@@ -988,7 +988,7 @@ func (runner *TestFileRunner) Cleanup(file *moduletest.File) {
 // Crucially, it differs from prepareInputVariablesForAssertions in that it only
 // includes variables that are reference by the config and not everything that
 // is defined within the test run block and test file.
-func buildInputVariablesForTest(run *moduletest.Run, file *moduletest.File, config *configs.Config, globals map[string]backend.UnparsedVariableValue) (tofu.InputValues, tfdiags.Diagnostics) {
+func buildInputVariablesForTest(run *moduletest.Run, file *moduletest.File, config *configs.Config, globals map[string]backend.UnparsedVariableValue, states map[string]*TestFileState) (tofu.InputValues, tfdiags.Diagnostics) {
```

Comment:
I talked this through with @cube2222, 

And we have a couple ideas about how to best approach this.

1. Construct the context at the top of this method, and evaluate each variable as we come across it. Then store a literal reference (evaluated) instead of an unknown value.

2. re-use the type for testVariableValueExpression in pr #1129 instead of tacking the evaluation context onto this type.

Which do you think is the better approach?

Reply:
Yes, I think that makes sense.

Though I'd still recommend creating the evaluation context before the for loop and reusing it across testVariableValueExpression's

Reply:
Done.